### ISSUE TITLE
Kafka crashes on startup with default values

### ISSUE THREAD

**Issue by invertigo** [bitnami/charts#3572]
**Which chart**:
bitnami/kafka-11.8.2

**Describe the bug**
At launch, the `kafka-0` pod goes into a CrashLoopBackoff with the following error:

```
[2020-08-31 22:19:43,303] INFO [ZooKeeperClient Kafka server] Connected. (kafka.zookeeper.ZooKeeperClient)
[2020-08-31 22:19:43,351] ERROR Fatal error during KafkaServer startup. Prepare to shutdown (kafka.server.KafkaServer)
org.apache.zookeeper.KeeperException$NoAuthException: KeeperErrorCode = NoAuth for /consumers
        at org.apache.zookeeper.KeeperException.create(KeeperException.java:120)
        at org.apache.zookeeper.KeeperException.create(KeeperException.java:54)
        at kafka.zookeeper.AsyncResponse.maybeThrow(ZooKeeperClient.scala:564)
        at kafka.zk.KafkaZkClient.createRecursive(KafkaZkClient.scala:1646)
        at kafka.zk.KafkaZkClient.makeSurePersistentPathExists(KafkaZkClient.scala:1568)
        at kafka.zk.KafkaZkClient.$anonfun$createTopLevelPaths$1(KafkaZkClient.scala:1560)
        at kafka.zk.KafkaZkClient.$anonfun$createTopLevelPaths$1$adapted(KafkaZkClient.scala:1560)
        at scala.collection.immutable.List.foreach(List.scala:392)
        at kafka.zk.KafkaZkClient.createTopLevelPaths(KafkaZkClient.scala:1560)
        at kafka.server.KafkaServer.initZkClient(KafkaServer.scala:445)
        at kafka.server.KafkaServer.startup(KafkaServer.scala:222)
        at kafka.server.KafkaServerStartable.startup(KafkaServerStartable.scala:44)
        at kafka.Kafka$.main(Kafka.scala:82)
        at kafka.Kafka.main(Kafka.scala)
```

**To Reproduce**
1. run `helm install --name kafka --namespace backend bitnami/kafka`

**Expected behavior**
Kafka is able to start

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
Client: &version.Version{SemVer:"v2.16.9", GitCommit:"8ad7037828e5a0fca1009dabe290130da6368e39", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.16.9", GitCommit:"8ad7037828e5a0fca1009dabe290130da6368e39", GitTreeState:"clean"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.4", GitCommit:"c96aede7b5205121079932896c4ad89bb93260af", GitTreeState:"clean", BuildDate:"2020-06-17T11:41:22Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.11-eks-065dce", GitCommit:"065dcecfcd2a91bd68a17ee0b5e895088430bd05", GitTreeState:"clean", BuildDate:"2020-07-16T01:44:47Z", GoVersion:"go1.12.17", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**

**Comment by joancafom** [comment]
Hi @invertigo ,

I have tried using default values but could not face any issues:

```console
$ helm install alpha bitnami/kafka
NAME: alpha
LAST DEPLOYED: Tue Sep  1 07:35:44 2020
STATUS: deployed
...

$ helm list
NAME 	NAMESPACE	REVISION	UPDATED                                	STATUS  	CHART       	APP VERSION
alpha	default.   	1       	2020-09-01 07:35:44.048163275 +0000 UTC	deployed	kafka-11.8.2	2.6.0

$ kubectl get pods
NAME                READY   STATUS    RESTARTS   AGE
alpha-kafka-0       1/1     Running   0          12m
alpha-zookeeper-0   1/1     Running   0          12m
```

I am trying to figure out why this may be happening... You are trying to create a brand new release (i.e. you don't have any previous PV with data), am I right?

**Comment by invertigo** [comment]
@joancafom - You were exactly right - it was trying to re-use some bad persistent volumes.  Thanks for the suggestion - I'm still new to working with statefulsets.

**Comment by Platou** [comment]
I have the exact same issue, but deleting PV/PVC didn't solve the issue and also I tried with a clean k8s cluster in vain.

**Comment by marcosbc** [comment]
@Platou Could you create a new issue with more details on how you are deploying your chart?

**Comment by Platou** [comment]
My issue was the same as above.

In the end, it was the `tls` vs `sasl_tls` properties. zookeeper is enforcing `sasl` and my kafka broker were configured with `tls` only therefore not providing sasl auth to zookeeper.

I configured them with `sasl_tls` to fix the issue.

**Comment by juan131** [comment]
Great @Platou ! I'm glad you were able to solve the issue after setting the right auth mechanism on Kafka & Zookeeper. Please do not hesitate to let us know if you require further help.

**Comment by cah-nitin-janavekar** [comment]
Hi Juan/Platou,

I am also facing the same issue on my GKE environment, could you please help let me know, which authority need to provide and how we bring back borker pod successfully.

[2021-11-24 11:06:08,523] INFO channel is connected: [id: 0xbb94909d, L:/10.58.38.191:40840 - R:ent-stg-cp-zookeeper/10.58.54.53:2281] (org.apache.zookeeper.ClientCnxnSocketNetty)
[2021-11-24 11:06:08,803] INFO Session establishment complete on server ent-stg-cp-zookeeper/10.58.54.53:2281, sessionid = 0x1003cfd65a90003, negotiated timeout = 18000 (org.apache.zookeeper.ClientCnxn)
[2021-11-24 11:06:08,806] INFO [ZooKeeperClient Kafka server] Connected. (kafka.zookeeper.ZooKeeperClient)
[2021-11-24 11:06:08,859] ERROR Fatal error during KafkaServer startup. Prepare to shutdown (kafka.server.KafkaServer)
org.apache.zookeeper.KeeperException$NoAuthException: KeeperErrorCode = NoAuth for /consumers
        at org.apache.zookeeper.KeeperException.create(KeeperException.java:120)
        at org.apache.zookeeper.KeeperException.create(KeeperException.java:54)
        at kafka.zookeeper.AsyncResponse.maybeThrow(ZooKeeperClient.scala:569)
        at kafka.zk.KafkaZkClient.createRecursive(KafkaZkClient.scala:1644)
        at kafka.zk.KafkaZkClient.makeSurePersistentPathExists(KafkaZkClient.scala:1566)
        at kafka.zk.KafkaZkClient.$anonfun$createTopLevelPaths$1(KafkaZkClient.scala:1558)
        at kafka.zk.KafkaZkClient.$anonfun$createTopLevelPaths$1$adapted(KafkaZkClient.scala:1558)
        at scala.collection.immutable.List.foreach(List.scala:392)
        at kafka.zk.KafkaZkClient.createTopLevelPaths(KafkaZkClient.scala:1558)
        at kafka.server.KafkaServer.initZkClient(KafkaServer.scala:425)
        at kafka.server.KafkaServer.startup(KafkaServer.scala:231)
        at io.confluent.support.metrics.SupportedServerStartable.startup(SupportedServerStartable.java:114)
        at io.confluent.support.metrics.SupportedKafka.main(SupportedKafka.java:66)
[2021-11-24 11:06:08,861] INFO shutting down (kafka.server.KafkaServer)
[2021-11-24 11:06:08,863] INFO [ZooKeeperClient Kafka server] Closing. (kafka.zookeeper.ZooKeeperClient)
[2021-11-24 11:06:08,870] INFO channel is told closing (org.apache.zookeeper.ClientCnxnSocketNetty)
[2021-11-24 11:06:08,872] INFO channel is disconnected: [id: 0xbb94909d, L:/10.58.38.191:40840 ! R:ent-stg-cp-zookeeper/10.58.54.53:2281] (org.apache.zookeeper.ClientCnxnSocketNetty)
[2021-11-24 11:06:08,872] INFO channel is told closing (org.apache.zookeeper.ClientCnxnSocketNetty)
[2021-11-24 11:06:08,872] INFO Session: 0x1003cfd65a90003 closed (org.apache.zookeeper.ZooKeeper)
[2021-11-24 11:06:08,872] INFO EventThread shut down for session: 0x1003cfd65a90003 (org.apache.zookeeper.ClientCnxn)
[2021-11-24 11:06:08,875] INFO [ZooKeeperClient Kafka server] Closed. (kafka.zookeeper.ZooKeeperClient)
[2021-11-24 11:06:08,879] INFO shut down completed (kafka.server.KafkaServer)
[2021-11-24 11:06:08,879] INFO Shutting down SupportedServerStartable (io.confluent.support.metrics.SupportedServerStartable)
[2021-11-24 11:06:08,879] INFO Closing BaseMetricsReporter (io.confluent.support.metrics.BaseMetricsReporter)
[2021-11-24 11:06:08,879] INFO Waiting for metrics thread to exit (io.confluent.support.metrics.SupportedServerStartable)
[2021-11-24 11:06:08,879] INFO Shutting down KafkaServer (io.confluent.support.metrics.SupportedServerStartable)
[2021-11-24 11:06:08,880] INFO shutting down (kafka.server.KafkaServer)

**Comment by juan131** [comment]
HI @cah-nitin-janavekar 

Could you please provide the values you're using to install the chart and the version of the chart you're using? Thanks in advance.

**Comment by cah-nitin-janavekar** [comment]
Hi Juan,

Thanks for quick reply, I am using community edition of kafka version 5.5.1 and we applied mTLS setup on my Kafka cluster in last year. But when I am renewing my kafka broker with new certificate, getting above error. Also, I have seen slight difference on certificate DN names on new and old certificate, I suspect might be, I am getting this error .
=================================================================================
                 ********************* kafka values.yaml ******************
# Default values for cp-kafka.
# This is a YAML-formatted file.
# Declare variables to be passed into your templates.

## ------------------------------------------------------
## Kafka
## ------------------------------------------------------
namespace: ent-kafka-stg

## Number of Kafka brokers
brokers: 3

## Image Info
## ref: https://hub.docker.com/r/confluentinc/cp-kafka/
image: nexus.abc.net:8445/confluentinc/cp-kafka
imageTag: 5.5.1

## Specify a imagePullPolicy
## ref: http://kubernetes.io/docs/user-guide/images/#pre-pulling-images
imagePullPolicy: IfNotPresent

## Specify an array of imagePullSecrets.
## Secrets must be manually created in the namespace.
## ref: https://kubernetes.io/docs/concepts/containers/images/#specifying-imagepullsecrets-on-a-pod
imagePullSecrets:

## StatefulSet Config
## Start and stop pods in Parallel or OrderedReady (one-by-one.)
## ref: https://kubernetes.io/docs/tutorials/stateful-application/basic-stateful-set/#pod-management-policy
podManagementPolicy: OrderedReady

## The StatefulSet Update Strategy which Kafka will use when changes are applied: OnDelete or RollingUpdate
## ref: https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies
updateStrategy: RollingUpdate

## Kafka Server properties
## ref: https://kafka.apache.org/documentation/#configuration
configurationOverrides:
  "offsets.topic.replication.factor": "3"
  "default.replication.factor": "3"
  "min.insync.replicas": 2
  "auto.create.topics.enable": true  ## Cannnot be set to False as schema registry and rest proxy will not work
  # "connections.max.idle.ms" : 60000
  # "log.flush.interval.ms" : 60000

  ## Options required for external access via NodePort
  ## ref:
  ## - http://kafka.apache.org/documentation/#security_configbroker
  ## - https://cwiki.apache.org/confluence/display/KAFKA/KIP-103%3A+Separation+of+Internal+and+External+traffic
  ##
  ## Advertised listeners will use the firstListenerPort value as it's default unless overridden here.
  ## Setting "advertised.listeners" here appends to "PLAINTEXT://${POD_IP}:9092,"

  "security.inter.broker.protocol": SSL
  "advertised.listeners": |-
   SSL://ent-stg-cp-kafka-${KAFKA_BROKER_ID}:9093,EXTERNAL://ent.kafka.brk${KAFKA_BROKER_ID}.stg.abc.net:9094,ISTIO://brk${KAFKA_BROKER_ID}.ent.kafka.stg.gcp.net:31400
  "listener.security.protocol.map": |- 
   PLAINTEXT:PLAINTEXT,SSL:SSL,EXTERNAL:SSL,ISTIO:SSL
  "authorizer.class.name" : "kafka.security.authorizer.AclAuthorizer"
  "super.users" : User:CN=kafka.server.stg.gcp.np.net,O=ABC,L=Dubai,ST=Dubai,C=UAE
  "allow.everyone.if.no.acl.found": "false"
  "ssl.truststore.location" : /etc/kafka/secrets/kafka.stg.server.truststore.jks
  "ssl.keystore.location" : /etc/kafka/secrets/kafka.stg.server.keystore.jks
  "ssl.endpoint.identification.algorithm": ''
  "ssl.client.auth": required
  
  ## mTLS-to-ZooKeeper

  zookeeper.connect : ent-stg-cp-zookeeper:2281  ## Zookeeper LoadBalancer
  zookeeper.connection.timeout.ms : 30000
  zookeeper.sync.time.ms : 30000
  zookeeper.ssl.client.enable : true
  zookeeper.client.Cnxn.Socket : org.apache.zookeeper.ClientCnxnSocketNetty
  zookeeper.client.secure : true
  zookeeper.ssl.protocol : SSL
  zookeeper.ssl.enabled.protocols : SSL
  zookeeper.ssl.keystore.location : /etc/kafka/secrets/kafka.stg.client.keystore.jks
  zookeeper.ssl.truststore.location : /etc/kafka/secrets/kafka.stg.client.truststore.jks
  zookeeper.ssl.endpoint.identification.algorithm : ''
  zookeeper.set.acl : true

     
passwordMap: 
    KAFKA_SSL_KEYSTORE_PASSWORD : "kafka-stg-server-keystore-creds"
    KAFKA_SSL_TRUSTSTORE_PASSWORD : "kafka-stg-server-truststore-creds"
    KAFKA_SSL_KEY_PASSWORD : "kafka-stg-server-key-creds"
    KAFKA_ZOOKEEPER_SSL_KEYSTORE_PASSWORD : "kafka-stg-client-keystore-creds"
    KAFKA_ZOOKEEPER_SSL_TRUSTSTORE_PASSWORD : "kafka-stg-client-truststore-creds"
    KAFKA_ZOOKEEPER_SSL_KEY_PASSWORD : "kafka-stg-client-key-creds"

## Additional env variables
customEnv: 
  # this section added - ent-stg
  #  KAFKA_SASL_SECRETS_DIR : "/etc/kafka/secrets"
    KAFKA_SSL_KEYSTORE_FILENAME : "kafka.stg.server.keystore.jks"
    KAFKA_SSL_KEYSTORE_CREDENTIALS : "kafka-stg-server-key-creds"
    KAFKA_SSL_TRUSTSTORE_FILENAME : "kafka.stg.server.truststore.jks"
    KAFKA_SSL_TRUSTSTORE_CREDENTIALS : "kafka-stg-server-truststore-creds"
    KAFKA_SSL_KEY_CREDENTIALS : "kafka-stg-server-key-creds"
    KAFKA_LOG4J_OPTS : "-Dlog4j.configuration=file:/etc/kafka/secrets/log4j.properties"
secrets:
  scrts : ent-stg-kafka-scrts-v001

persistence:
  enabled: true

  ## The size of the PersistentVolume to allocate to each Kafka Pod in the StatefulSet. For
  ## production servers this number should likely be much larger.
  size: 500Gi

  ## Kafka data Persistent Volume Storage Class
  ## If defined, storageClassName: <storageClass>
  ## If set to "-", storageClassName: "", which disables dynamic provisioning
  ## If undefined (the default) or set to null, no storageClassName spec is
  ##   set, choosing the default provisioner.  (gp2 on AWS, standard on
  ##   GKE, AWS & OpenStack)
  ##
  storageClass: "standard-regional"

  disksPerBroker: 1

## Kafka JVM Heap Option
heapOptions: "-Xms512M -Xmx10240M"

resources: {}
  # We usually recommend not to specify default resources and to leave this as a conscious
  # choice for the user. This also increases chances charts run on environments with little
  # resources, such as Minikube. If you do want to specify resources, uncomment the following
  # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
  # limits:
  #  cpu: 100m
  #  memory: 128Mi
  # requests:
  #  cpu: 100m
  #  memory: 128Mi

## Custom pod annotations
podAnnotations: {}

## Node labels for pod assignment
## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
nodeSelector:
    run: kafka 

## Taints to tolerate on node assignment:
## Ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
tolerations: 
  - key: run
    operator: Equal
    value: kafka
    effect: NoSchedule

## Monitoring
## Kafka JMX Settings
## ref: https://docs.confluent.io/current/kafka/monitoring.html
jmx:
  port: 5555

## Prometheus Exporter Configuration
## ref: https://prometheus.io/docs/instrumenting/exporters/
prometheus:
  ## JMX Exporter Configuration
  ## ref: https://github.com/prometheus/jmx_exporter
  jmx:
    enabled: true
    image: solsson/kafka-prometheus-jmx-exporter@sha256
    imageTag: 6f82e2b0464f50da8104acd7363fb9b995001ddff77d248379f8788e78946143
    imagePullPolicy: IfNotPresent
    port: 5556

    ## Resources configuration for the JMX exporter container.
    ## See the `resources` documentation above for details.
    resources: {}

nodeport:
  enabled: false
  servicePort: 19092
  firstListenerPort: 31090

## ------------------------------------------------------
## Zookeeper
## ------------------------------------------------------
cp-zookeeper:
  ## If true, install the cp-zookeeper chart alongside cp-kafka
  ## ref: ../cp-zookeeper
#  enabled: true
#  servers: 3
#  persistence:
#    enabled: true
#    dataDirSize: 20Gi
#    dataLogDirSize: 50Gi

  ## If the Zookeeper Chart is disabled a URL and port are required to connect
  url: "ent-stg-cp-zookeeper:2281"
# Ingress
service:
  type: LoadBalancer
ingress:
  enabled: true
  host: brk.ent.kafka.stg.gcp.cahcloud.net
  host0: brk0.ent.kafka.stg.gcp.cahcloud.net
  host1: brk1.ent.kafka.stg.gcp.cahcloud.net
  host2: brk2.ent.kafka.stg.gcp.cahcloud.net

**Comment by rafariossaa** [comment]
Hi,
  Could you try with bitnami's images ? The chart and settings and meant to work with bitnami's images. If you use other images, maybe they don't initialize in the same way.
